wasm-chain-integration/src/v1/tests.rs

```diff
@@ -0,0 +1,839 @@
+use super::{trie, types::*};
+use crate::v1::trie::MutableState;
+use anyhow::{bail, ensure};
+use quickcheck::*;
+
+const NUM_TESTS: u64 = 100000;
+
+#[test]
+/// This tests performs the following tasks:
+/// 1. Entries can be created.
+/// 2. Created entries can be looked up.
+/// 3. Writing to created entries and checking that the returned 'write length'
+/// is accurate.
+/// 4. Reading from an invalidated entry should return u32::MAX.
+/// 5. Writing to an invalidated entry should return u32::MAX.
+/// 6. Entries can be deleted check that equivalent entries are also
+/// invalidated. 7. Deleting already deleted entries returns u32::MAX
+fn prop_create_write_read_delete() {
+    let prop = |inputs: Vec<(Vec<u8>, trie::Value)>| -> anyhow::Result<()> {
+        let loader = trie::Loader {
+            inner: Vec::<u8>::new(),
+        };
+        let mut m_state = MutableState::initial_state();
+        let mut state = InstanceState::new(0, loader, m_state.get_inner());
+        for (k, v) in &inputs {
+            let entry = if let Ok(entry) = state.create_entry(k) {
+                entry
+            } else {
+                bail!("The entry should've been created {:?}.", k);
+            };
+            let entry = if let Some(entry) = entry.convert() {
+                entry
+            } else {
+                bail!("Entry should be valid.");
+            };
+
+            let lookup_entry = if let Some(lookup_entry) = state.lookup_entry(k).convert() {
+                lookup_entry
+            } else {
+                bail!("Lookup entry should be valid.");
+            };
+
+            let write_res = if let Ok(write_res) = state.entry_write(entry, v, 0) {
+                write_res
+            } else {
+                bail!("Failed writing {:?} to key {:?}.", v, k);
+            };
+
+            ensure!(
+                write_res as usize == v.len(),
+                "Entry write returned wrong write length {:?} expected {:?}.",
+                write_res,
+                v.len()
+            );
+
+            let mut buff0 = vec![0; v.len()];
+            let read0 = state.entry_read(entry, &mut buff0, 0);
+            ensure!(
+                read0 as usize == v.len(),
+                "Unexpected read length {:?} expected {:?}.",
+                read0,
+                v.len()
+            );
+            let read1 = state.entry_read(lookup_entry, &mut buff0, 0);
+            ensure!(
+                read1 as usize == v.len(),
+                "Unexpected read length {:?} expected {:?}.",
+                read1,
+                v.len()
+            );
+            for _ in 0..1 {
```

Comment:
This is a strange for loop. What is the reason for the loop over a singleton, where the variable is not used?

Reply:
Testing the reviewer ... :) 

No I think it is the remains of some prior version of the test. 
I will remove it.

Reply:
Phew, passed the test then :)